src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java

```diff
@@ -47,6 +50,19 @@ private HttpSender getHttpSender() {
 
 	@Override
 	protected Void doInBackground() throws Exception {
+		TreeSet<HtmlParameter> cookies = this.httpMessage.getRequestHeader().getCookieParams();
+		if (targetToken.getType() == HtmlParameter.Type.cookie) {
+			Iterator<HtmlParameter> iter = cookies.iterator();
+			while (iter.hasNext()) {
+				HtmlParameter cookie = iter.next();
+				if (cookie.getName().equals(targetToken.getName())) {
+					iter.remove();
+					break;
+				}
+			}
+		}
+		HttpMessage msg = this.httpMessage.cloneRequest();
+		msg.getRequestHeader().setCookieParams(cookies);

```

Comment:
I understand that use case, but not removing the cookies will break other use cases. For example, it would no longer be possible to check a form field (e.g. `authenticity_token`) that is bound to the session.
Probably better to allow the user to choose when the cookies are removed or not.